docs/design/clustering.md

```diff
@@ -0,0 +1,56 @@
+# Clustering in Kubernetes
+
+
+## Overview
+The term "clustering" refers to the process of having all members of the kubernetes cluster find and trust each other.  There are multiple different ways to achieve clustering with different security and usability profiles.  This document attempts to lay out the user experiences for clustering that Kubernetes aims to address.
+
+Once a cluster is established, the following is true:
+
+1. **Master -> Node**  The master needs to know which nodes can take work and what their current status is wrt capacity.
+  1. **Location** The master knows the name and location of all of the nodes in the cluster.
```

Comment:
Do we want to state here that for now, the master must reach the node via a pathway that the node can reach itself?  Or do we want to bake in the distinction that what node thinks of as its name may not be how the master reaches it?  I would prefer the former, but expect people to ask about the latter.